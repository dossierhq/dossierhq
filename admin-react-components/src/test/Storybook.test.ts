import initStoryshots, { Stories2SnapsConverter } from '@storybook/addon-storyshots';
import { act, create, ReactTestRenderer } from 'react-test-renderer';

const converter = new Stories2SnapsConverter();

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const runTest = async (story: any, context: any) => {
  const filename = converter.getSnapshotFileName(context);

  if (!filename) {
    return;
  }

  const storyElement = story.render();
  let tree: ReactTestRenderer | undefined;
  await act(async () => {
    tree = create(storyElement);
  });

  expect(tree?.toJSON()).toMatchSpecificSnapshot(filename);

  tree?.unmount();
};

initStoryshots({
  asyncJest: true,
  test: ({ story, context, done }) => {
    runTest(story, context).then(done);
  },
});
