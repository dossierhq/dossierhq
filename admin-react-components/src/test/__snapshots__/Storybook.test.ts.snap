// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Button Disabled 1`] = `
<button
  className="dd button text-button "
  disabled={true}
  type="button"
>
  Text
</button>
`;

exports[`Storyshots Components/Button Normal 1`] = `
<button
  className="dd button text-button "
  type="button"
>
  Text
</button>
`;

exports[`Storyshots Components/Button Primary 1`] = `
<button
  className="dd button text-button bg-primary"
  type="button"
>
  Text
</button>
`;

exports[`Storyshots Components/Button Primary Disabled 1`] = `
<button
  className="dd button text-button bg-primary"
  disabled={true}
  type="button"
>
  Text
</button>
`;

exports[`Storyshots Components/EntityEditor Deleted Foo 1`] = `
<form
  className="dd form has-background"
  onSubmit={[Function]}
>
  <div>
    <label
      className="dd text-subtitle1"
      htmlFor="entity-fc66b4d7-61ff-44d4-8f68-cb7f526df046-_name"
    >
      Name
    </label>
    <div
      className="dd form-control"
    >
      <input
        className="dd text-body1 input"
        id="entity-fc66b4d7-61ff-44d4-8f68-cb7f526df046-_name"
        onChange={[Function]}
        type="text"
        value="Hello"
      />
    </div>
  </div>
  <hr
    className="dd divider"
  />
  <div>
    <label
      className="dd text-subtitle1"
      htmlFor="entity-fc66b4d7-61ff-44d4-8f68-cb7f526df046-title"
    >
      title
    </label>
    <div
      className="dd form-control"
    >
      <input
        className="dd text-body1 input"
        id="entity-fc66b4d7-61ff-44d4-8f68-cb7f526df046-title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <input
    className="dd button text-button bg-primary"
    disabled={false}
    type="submit"
    value="Save"
  />
</form>
`;

exports[`Storyshots Components/EntityEditor Full Foo 1`] = `
<form
  className="dd form has-background"
  onSubmit={[Function]}
>
  <div>
    <label
      className="dd text-subtitle1"
      htmlFor="entity-fc66b4d7-61ff-44d4-8f68-cb7f526df046-_name"
    >
      Name
    </label>
    <div
      className="dd form-control"
    >
      <input
        className="dd text-body1 input"
        id="entity-fc66b4d7-61ff-44d4-8f68-cb7f526df046-_name"
        onChange={[Function]}
        type="text"
        value="Hello"
      />
    </div>
  </div>
  <hr
    className="dd divider"
  />
  <div>
    <label
      className="dd text-subtitle1"
      htmlFor="entity-fc66b4d7-61ff-44d4-8f68-cb7f526df046-title"
    >
      title
    </label>
    <div
      className="dd form-control"
    >
      <input
        className="dd text-body1 input"
        id="entity-fc66b4d7-61ff-44d4-8f68-cb7f526df046-title"
        onChange={[Function]}
        type="text"
        value="Hello"
      />
    </div>
  </div>
  <input
    className="dd button text-button bg-primary"
    disabled={false}
    type="submit"
    value="Save"
  />
</form>
`;

exports[`Storyshots Components/EntityEditor New Foo 1`] = `
<form
  className="dd form has-background"
  onSubmit={[Function]}
>
  <div>
    <label
      className="dd text-subtitle1"
      htmlFor="new-entity-123-_name"
    >
      Name
    </label>
    <div
      className="dd form-control"
    >
      <input
        className="dd text-body1 input"
        id="new-entity-123-_name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <hr
    className="dd divider"
  />
  <div>
    <label
      className="dd text-subtitle1"
      htmlFor="new-entity-123-title"
    >
      title
    </label>
    <div
      className="dd form-control"
    >
      <input
        className="dd text-body1 input"
        id="new-entity-123-title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <input
    className="dd button text-button bg-primary"
    disabled={true}
    type="submit"
    value="Save"
  />
</form>
`;

exports[`Storyshots Components/HelloWorld Normal 1`] = `
<div>
  Hello world
</div>
`;

exports[`Storyshots Components/InputSubmit Disabled 1`] = `
<input
  className="dd button text-button bg-primary"
  disabled={true}
  type="submit"
  value="Done"
/>
`;

exports[`Storyshots Components/InputSubmit Normal 1`] = `
<input
  className="dd button text-button bg-primary"
  type="submit"
  value="Done"
/>
`;
